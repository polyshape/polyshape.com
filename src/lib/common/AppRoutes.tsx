import { Outlet } from "react-router-dom";
import Home from "../../pages/Home";
import Vision from "../../pages/orientation/Vision";
import Journey from "../../pages/orientation/Journey";
import Software from "../../pages/r_n_d/Software";
import Patents from "../../pages/r_n_d/Patents";
import Publications from "../../pages/r_n_d/Publications";
import Dissemination from "../../pages/r_n_d/Dissemination";
import Projects from "../../pages/portfolio/Projects";
import Partners from "../../pages/portfolio/Partners";
import News from "../../pages/News";
import Contact from "../../pages/Contact";
import type { JSX } from "react";

export const AppRoutes = {
  HOME: {
    id: "home",
    path: "/",
    title: "Home",
    element: <Home/>,
    isParent: true,
    children: [] as const
  },
  ORIENTATION: {
    id: "orientation",
    path: "/orientation",
    title: "Orientation",
    element: <Outlet/>,
    isParent: true,
    children: ["VISION", "JOURNEY"] as const
  },
  VISION: {
    id: "vision",
    path: "/orientation/vision",
    title: "Vision",
    element: <Vision/>,
    isParent: false,
    children: [] as const
  },
  JOURNEY: {
    id: "journey",
    path: "/orientation/journey",
    title: "Journey",
    element: <Journey/>,
    isParent: false,
    children: [] as const
  },
  RND: {
    id: "rnd",
    path: "/rnd",
    title: "R&D",
    element: <Outlet/>,
    isParent: true,
    children: ["SOFTWARE", "PATENTS", "PUBLICATIONS", "DISSEMINATION"] as const
  },
  SOFTWARE: {
    id: "software",
    path: "/rnd/software",
    title: "Software",
    element: <Software/>,
    isParent: false,
    children: [] as const
  },
  PATENTS: {
    id: "patents",
    path: "/rnd/patents",
    title: "Patents",
    element: <Patents/>,
    isParent: false,
    children: [] as const
  },
  PUBLICATIONS: {
    id: "publications",
    path: "/rnd/publications",
    title: "Publications",
    element: <Publications/>,
    isParent: false,
    children: [] as const
  },
  DISSEMINATION: {
    id: "dissemination",
    path: "/rnd/dissemination",
    title: "Dissemination",
    element: <Dissemination/>,
    isParent: false,
    children: [] as const
  },
  PORTFOLIO: {
    id: "portfolio",
    path: "/portfolio",
    title: "Portfolio",
    element: <Outlet/>,
    isParent: true,
    children: ["PROJECTS","PARTNERS"] as const
  },
  PROJECTS: {
    id: "projects",
    path: "/portfolio/projects",
    title: "Projects",
    element: <Projects/>,
    isParent: false,
    children: [] as const
  },
  PARTNERS: {
    id: "partners",
    path: "/portfolio/partners",
    title: "Partners",
    element: <Partners/>,
    isParent: false,
    children: [] as const
  },
  NEWS: {
    id: "news",
    path: "/news",
    title: "News",
    element: <News/>,
    isParent: true,
    children: [] as const
  },
  CONTACT: {
    id: "contact",
    path: "/contact",
    title: "Contact",
    element: <Contact/>,
    isParent: true,
    children: [] as const
  }
} as const;

export type AppRoute = {
  id: string;
  path: string;
  title: string;
  element: JSX.Element;
  isParent: boolean;
  children: readonly (keyof typeof AppRoutes)[];
};
