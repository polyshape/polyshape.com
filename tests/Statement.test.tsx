import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import Statement from '../src/pages/mission/Statement';
import { AppRoutes } from '../src/lib/common/AppRoutes';

describe('Statement page', () => {
  it('renders the main title', () => {
    const { getByText } = render(<Statement />);
    expect(getByText(AppRoutes.STATEMENT.title)).toBeInTheDocument();
  });

  it('applies correct CSS classes', () => {
    const { container } = render(<Statement />);
    expect(container.querySelector('.prose')).toBeInTheDocument();
    expect(container.querySelector('.hero__title')).toBeInTheDocument();
  });

  it('matches the snapshot', () => {
    const { container } = render(<Statement />);
    expect(container).toMatchSnapshot();
  });
});
