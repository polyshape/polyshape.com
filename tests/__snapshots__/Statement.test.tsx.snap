// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Statement page matches the snapshot 1`] = `
<div>
  <div
    class="prose"
  >
    <h1
      class="hero__title"
    >
      Mission Statement
    </h1>
    <p>
      The future of human health demands the best tools we have—and there is no tool more powerful, more transformative, than Artificial General Intelligence (AGI). To truly support human health, we must be bold enough to tap into AGI’s full potential, unlocking a new era of scientific discovery, accelerated innovation, and holistic well-being.
    </p>
    <p>
      AGI stands as the ultimate catalyst for health-driven breakthroughs. From early detection and precise diagnostics to safeguarding individuals and accelerating drug discovery, its capabilities span the entire spectrum of healthcare. But beyond that, AGI has the power to elevate human potential, enhancing mental well-being, personalizing care, and ensuring that no one is left behind in the pursuit of health and longevity.
    </p>
    <p>
      Our vision is Generalized Health Intelligence, a revolution in AI-driven healthcare. Imagine an interconnected Matrix of AGI agents, swarming ahead of us, for us—autonomous, intelligent, and proactive in identifying risks, optimizing treatments, and supporting well-being at every stage of life. These agents do not replace human expertise; they empower it. They amplify human creativity, safeguard health, and accelerate the development of life-saving innovations at an unprecedented scale.
    </p>
    <p>
      While ethics and regulation are essential, we embrace them with a spirit of positive transformation. When harnessed responsibly, AGI is not a threat, it is an infinite opportunity. It is the key to a future where health is not just treated but predicted, prevented, and optimized through intelligence far beyond what any single human mind can achieve.
    </p>
    <p>
      We are not here to claim the future, we are here to build it. Science moves forward through careful reasoning, responsible innovation, and an unwavering commitment to human well-being. AGI is no different. It expands human insight, accelerating what once seemed impossible. If done right, it will not just transform healthcare but redefine how we understand health itself.
    </p>
  </div>
</div>
`;
